lib/streamlit/components/v1/components.py

```diff
@@ -109,6 +108,34 @@ def create_instance(
         if len(args) > 0:
             raise MarshallComponentException(f"Argument '{args[0]}' needs a label")
 
+        try:
+            import pyarrow
+            from streamlit.elements import arrow_table
+        except ImportError:
+            import sys
+
+            if sys.version_info >= (3, 9):
+                raise StreamlitAPIException(
+                    """To use Custom Components in Streamlit, you need to install
+PyArrow. Unfortunately, PyArrow does not yet support Python 3.9.
+
+You can either switch to Python 3.8 with an environment manager like PyEnv, or
+[install Streamlit with conda](https://discuss.streamlit.io/t/note-installation-issues-with-python-3-9-and-streamlit/6946/7):
```

Comment:
@akrolsmir : I'm a little confused by reading this line. Are we telling the user to 

1. install Python 3.8 (and streamlit) using conda as en environment manager 
OR 
2. are we saying that by using conda the user can use Python 3.9 and PyArrow

Can we make the distinction clear in the comment?

Reply:
install Python 3.8 (and streamlit) using any environment manager or use conda

Reply:
Cool, LGTM 👍